### ISSUE TITLE
Smart Life Device not displayed in new Tuya Plugin

### ISSUE THREAD

**Issue by Vector-Int13** [tuya/tuya-home-assistant#569]
**Describe the bug**
Smart Life device appears on the IoT Web, the app is paired. Tuya and Smart Life.

I'll log into the plugin. Plugin runs, connects, but only Tuya devices are visible in the plugin and no Smart Life.

**Expected behavior**
I tried removing Tuya pairing and leaving only Smart Life in IOT WEB, but then the plugin failed to log in.

**Home Assistant Version**
- Home Assistant 2021.10.3

Before the plugin change, I was only using Smart Life devices.  I only have a few pieces of equipment and I was completely comfortable with the way the original plugin worked.
I find the discontinuation without notice and any functional replacement shameful.
There is no separate smart life plugin as far as I know. Neither is there an option to control these devices via LAN.  Tuya LAN protocol does not match either.

**Comment by ChrisEdwardsCE** [comment]
Agreed. They rolled this out when it was not even close to done. For me, none of my Tuya devices are supported.

**Comment by trinityr** [comment]
My issue was resolved by removing the user I created in the cloud application.  When you connect to your cloud account use your Smart Life application information.

**Comment by Trukmeister** [comment]
@trinityr I used the Tuya Smart application information to log in from Home Assistant and used the keys for the Tuya IoT Development account but, when you say "removing the user I created in the cloud application" I am not sure exactly what you mean.  Do you mean the IoT Tuya Development account?  If you would not mind elaborating a little more please and thanks!!  I am completely stuck.  Everything from the Tuya Smart app works great just nothing shows in the HA.

**Comment by Trukmeister** [comment]
Okay weird!  I got it working but after a lot of messing around.  I installed the latest core-2021.10.7 - removed the Tuya integration and add it back again. Did not work.  I never could get the integration to connect unless I added an authorized user to the IoT project - then it would log me in (this denial of access and then access granted AFTER granting my Tuya smart life account access to the iOT project happened several times).  However that is not how it works or what fixed the issue.  The Tuya integration documentation does not show granting a user/account access to the IoT Project.  So I decided to try one more time (keeping in mind the newest core was installed which MAY have been the magic bullet).  I deleted all IoT projects and created another IoT Project.  I then uninstalled/removed the Tuya integration from HA.  Added to HA the Tuya integration again and used my new Authorization Key and Access Key and my Tuya Smart app username and password - keeping in mind I did not add an authorized user to the IoT Project.  Presto - it worked and I now have my Tuya devices showing in HA  :).  I am just a little happy now.  @frenck hope this was a little more positive of a post :) and in the future I will stick more to the local devices if at all possible.  We live and learn.

**Comment by Trukmeister** [comment]
Newly added devices are not showing up in HA but they do show in the device tab in the IoT Project. 
Enable newly added entities is enabled.  Reloaded the integration and restarted HA - still missing newly added switches.
any ideas?

**Comment by Vector-Int13** [comment]
I changed my email in my Smart Life account. 
Now I have a different email in Tuya and smart Life account and one of the Smart life devices has appeared after restarting the module.
I finally see the air conditioning on Smart Life platform.
Unfortunately, I still can't see the other two devices.
This is how I tried that the tuya_platform parameter is ignored!
It doesn't matter if you have Tuya or smart_life in it. He just tries to load Tuya first, and if he doesn't, smart life is loaded.
Absolutely terrible slender.
The original accessory worked excellently. This is not even at the BETA level. And above all, support for the original was never to end as a system upgrade that cannot be undone. That's unbelievable.
I hope that my knowledge will help someone load at least some of their smart life devices.
Unfortunately, the add-on on HomeBridge behaves the same way, so it will be an API property.

**Comment by Vector-Int13** [comment]
> 
> 
> @Trukmeister Maybe I've been personally investing tons of time these past weeks to address issues and spent a shitload amount of money on a lot of Tuya devices, simply to help the community out. Time spent well I must say, responses to the beta are pretty OK up until now. I don't care about Tuya personally, I don't use it; I choose my device carefully upfront (local). But, yeah, what you are doing, spreading negativity and being a total jackass online, is also really helpful and productive for the community.
> 
> Thank you, people that just do that, like yourself in the above messages, and their approach really motivates me to do something good for the community (to be sure: that is sarcasm).
> 
> Thank you, for your contributions!

I can't help but remind me of a man handing out soup in the Salvation Army to the homeless. When the dead fall to the ground because she is poisoned, he tells them that it was free and cooked it in his spare time.

People who do not feel responsible and act as if the world belongs to them are poison for all open source projects.

**Comment by Vector-Int13** [comment]
> 
> 
> My issue was resolved by removing the user I created in the cloud application. When you connect to your cloud account use your Smart Life application information.

Thanks friend, you pointed me in the right direction. One mystery is now solved for me.

**Comment by Vector-Int13** [comment]
> 
> 
> @Trukmeister Maybe I've been personally investing tons of time these past weeks to address issues and spent a shitload amount of money on a lot of Tuya devices, simply to help the community out. Time spent well I must say, responses to the beta are pretty OK up until now. I don't care about Tuya personally, I don't use it; I choose my device carefully upfront (local). But, yeah, what you are doing, spreading negativity and being a total jackass online, is also really helpful and productive for the community.
> 
> Thank you, people that just do that, like yourself in the above messages, and their approach really motivates me to do something good for the community (to be sure: that is sarcasm).
> 
> Thank you, for your contributions!

I was looking at a new add-on in 11.1. Good work! Although the mobile air conditioning still doesn't work properly for me, he saw that you did a lot of work.
I also read an explanation of how Tuya changed the policy of ending support.
I mean, I understand that it's not easy to work in such conditions.
I still consider removing the old version to be a mistake, but I want to thank you for your work.

**Comment by Trukmeister** [comment]
> @Trukmeister Maybe I've been personally investing tons of time these past weeks to address issues and spent a shitload amount of money on a lot of Tuya devices, simply to help the community out. Time spent well I must say, responses to the beta are pretty OK up until now. I don't care about Tuya personally, I don't use it; I choose my device carefully upfront (local). But, yeah, what you are doing, spreading negativity and being a total jackass online, is also really helpful and productive for the community.
> 
> Thank you, people that just do that, like yourself in the above messages, and their approach really motivates me to do something good for the community (to be sure: that is sarcasm).
> 
> Thank you, for your contributions!

@frenck I was patient :) my last missing devices showed up (dimmer switches) in the 2021.11.1 core update - thanks!!  Now my only problem (not related to anything you do) is with the Tuya IoT web page - I create an account and the next day I cannot log in and I cannot do the forgot account password even though I did not forget my password.  Something is up with their site.  Seriously thanks frenck!

**Comment by Trukmeister** [comment]
Is it just me or are the Tuya devices not available in the Automations?  I have tried to create automations with my Tuya dimmer light switches and they are not in the device list in Automations.  Other Tuya devices are in the list but not dimmer light switches.
